Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Cancel.hs

```diff
@@ -89,6 +94,26 @@ cancel req merchant booking = do
   QBCR.upsert bookingCR
   QRB.updateStatus booking.id SRB.CANCELLED
 
+  transporterConfig <- SCT.findByMerchantOpCityId booking.merchantOperatingCityId >>= fromMaybeM (TransporterConfigNotFound booking.merchantOperatingCityId.getId)
+  isChargable <- case (mbRide, mbDriverDistanceToPickup) of
+    (Just ride, Just driverDistanceToPickup) -> do
+      now <- getCurrentTime
+      let driverBookingDuration = roundToIntegral $ diffUTCTime now booking.createdAt
+      let condition = driverBookingDuration > transporterConfig.driverBookingTime && driverDistanceToPickup < transporterConfig.driverDistanceToPickupBound
+      if not condition
+        then do
+          driverQuote <- QDQ.findById (Id booking.quoteId) >>= fromMaybeM (InternalError ("Driver Quotes Not Found for quote id- " <> booking.quoteId))
+          searchReqForDriver <- QSRD.findByDriverSearchTryIdAndResponse ride.driverId driverQuote.searchTryId >>= fromMaybeM (InternalError "SearchRequestForDriver Not Found")
+          return $ (searchReqForDriver.actualDistanceToPickup - driverDistanceToPickup) > transporterConfig.driverPickupDistanceBound
+        else return condition
+    _ -> return False
+
+  when isChargable $ do
+    unless (isJust booking.riderId) $ do
+      throwError $ InternalError ("Rider Id does not exist for BookingId -" <> booking.id.getId)
```

Comment:
the same, don't use partial functions:

`riderId <-  booking.riderId & fromMaybeM (InternalError ("Rider Id does not exist for BookingId -" <> booking.id.getId))`